### DISCUSSION TITLE
brew update takes a long time now and cpu is maxed out

### DISCUSSION THREAD

**Discussion by adamwinn** [Homebrew/discussions#710] [Everyday usage]
When I run `brew update`, it takes a long time to complete the cpu is maxed out.  Processes like `tr`, and `git` are taking up 100% cpu

The 3 previous high cpu spikes in red are when I ran `brew update` earlier.  The red spike on the right of the graph is me currently running `brew update`:
![Screen Shot 2021-02-05 at 12 34 16 PM](https://user-images.githubusercontent.com/3028964/107080619-c7439e80-67ae-11eb-870c-e5ef8083233a.jpg)

Another screenshot:
![Screen Shot 2021-02-05 at 12 37 41 PM](https://user-images.githubusercontent.com/3028964/107080759-f22df280-67ae-11eb-84a0-995411a3e611.jpg)

```
brew config
HOMEBREW_VERSION: 3.0.0
ORIGIN: https://github.com/Homebrew/brew
HEAD: ebc7cc36a0e029d58fac8cebd1b5ff583b3fd95b
Last commit: 10 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 8bf0ee917c5b5f2de1aecec375f3ae37c70db5e4
Core tap last commit: 24 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_NO_AUTO_UPDATE: set
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit westmere
Clang: 12.0 build 1200
Git: 2.30.0 => /usr/local/bin/git
Curl: 7.64.1 => /usr/bin/curl
Java: 1.8.0_51
macOS: 11.2-x86_64
CLT: 12.4.0.0.1.1610135815
Xcode: N/A
XQuartz: 2.7.11 => /opt/X11
Rosetta 2: true
```
```
brew doctor
Your system is ready to brew.
```

**Comment by glensc** [comment]
@adamwinn it would help a lot if you show the commands that brew spawns, perhaps `git gc` is being invoked by git internally

```
$ ps uaww
```

or even better, save output to file, and grep only terminal the brew updae was ran:

```
$ ps uaww > ps.txt
$ grep s000 ps.txt
```
